src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java

```diff
@@ -378,25 +379,12 @@ private static boolean shouldNotifyUsersOnFeedUpdated(FeedSource formerFeedSourc
      * @return              A feed source containing only labels the user is allowed to see
      */
     protected static FeedSource cleanFeedSourceLabels(FeedSource feedSource, boolean isAdmin) {
-        // Admin can see all labels
-        if (isAdmin) {
-            feedSource.labelIds = Persistence.labels.getFiltered(
-                    in("_id", feedSource.labelIds)
-                    ).stream()
-                    .map(label -> label.id)
-                    .collect(Collectors.toList());
-        }
-        // Remove labels user is not allowed to see if user is not admin
-        if (!isAdmin) {
-            feedSource.labelIds = Persistence.labels.getFiltered(
-                    and(
-                            eq("adminOnly", false),
-                            in("_id", feedSource.labelIds)
-                    )).stream()
-                    .map(label -> label.id)
-                    .collect(Collectors.toList());;
-        }
-
+        Bson labelFilter = in("_id", feedSource.labelIds);
+        // Admin can view all feed labels, but a non-admin should only see those with adminOnly=false
+        Bson filter = isAdmin ? labelFilter : and(labelFilter, eq("adminOnly", false));
```

Comment:
This is really clever!

Reply:
Thanks! Even cleverer in https://github.com/ibi-group/datatools-server/commit/778b8b68f80bc82c572a75c422297abd87153b81#diff-798d3c9ae02799187190a03b96c88c8bc94ea435f80d4722e3684f9c674c5787R19-R21

Hopefully not too clever though?

Reply:
Looks really clear to me!